fastlane/Fastfile

```diff
@@ -189,62 +189,64 @@ ENV["HAS_ALPHA_VERSION"]="true"
   # build_release
   # -----------------------------------------------------------------------------------
-  # This lane builds the final release of the app and uploads it 
+  # This lane builds the final release of the app and uploads it
   # -----------------------------------------------------------------------------------
   # Usage:
   # bundle exec fastlane build_release [skip_confirm:<skip confirm>]
   #
   # Example:
-  # bundle exec fastlane build_release 
-  # bundle exec fastlane build_release skip_confirm:true 
+  # bundle exec fastlane build_release
+  # bundle exec fastlane build_release skip_confirm:true
   #####################################################################################
   desc "Builds and updates for distribution"
-  lane :build_release do | options |
-    android_build_prechecks(skip_confirm: options[:skip_confirm], 
+  lane :build_and_upload_release do | options |
+    android_build_prechecks(skip_confirm: options[:skip_confirm],
       alpha: false,
       beta: false,
       final: true)
     android_build_preflight()
-    
+
     # Create the file names
     version=android_get_release_version()
     build_bundle(version: version, flavor:"Vanilla", buildType: "release")
+
+    upload_build_to_play_store(version: version["name"], track: "production")
   end
 
   #####################################################################################
-  # build_pre_releases
+  # build_and_upload_pre_releases
   # -----------------------------------------------------------------------------------
-  # This lane builds the app it for both internal and external distribution 
+  # This lane builds the app it for both internal and external distribution and uploads them
   # -----------------------------------------------------------------------------------
   # Usage:
-  # bundle exec fastlane build_pre_releases [skip_confirm:<skip confirm>]
+  # bundle exec fastlane build_and_upload_pre_releases [skip_confirm:<skip confirm>]
   #
   # Example:
-  # bundle exec fastlane build_pre_releases 
-  # bundle exec fastlane build_pre_releases skip_confirm:true 
+  # bundle exec fastlane build_and_upload_pre_releases
+  # bundle exec fastlane build_and_upload_pre_releases skip_confirm:true
   #####################################################################################
   desc "Builds and updates for distribution"
-  lane :build_pre_releases do | options |
-    android_build_prechecks(skip_confirm: options[:skip_confirm], 
+  lane :build_and_upload_pre_releases do | options |
+    android_build_prechecks(skip_confirm: options[:skip_confirm],
       alpha: true,
       beta: true,
       final: false)
     android_build_preflight()
-    build_alpha(skip_prechecks: true, skip_confirm: options[:skip_confirm])
-    build_beta(skip_prechecks: true, skip_confirm: options[:skip_confirm])
+    build_alpha(skip_prechecks: true, skip_confirm: options[:skip_confirm], upload_to_play_store: true)
```

Comment:
One suggestion just based on the [WPiOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/Scripts/fastlane/Fastfile#L340) and [WooAndroid](https://github.com/woocommerce/woocommerce-android/blob/develop/fastlane/Fastfile#L227) implementations – in those cases we renamed the lanes to `build_and_upload_{type}` to better communicate what they do, and they always do the upload. Having an upload option on a `build` lane (which I kinda think of as a method) seems like it'd almost make more sense as its own lane _after_ building?

I think the best way to go here might just be to be consistent with WCAndroid?

Reply:
The reason I went this route is to give us the ability to build the alpha or beta without uploading to play store. I followed WCAndroid's example with the main lanes `build_and_upload_release` & `build_and_upload_pre_releases`. However, at times I have to try to build the app to fix build errors but I don't want it to be uploaded, so these lanes would be helpful. Having said that, they are not crucial either, I can just run the necessary commands manually. I just prefer this because I know it'll be the same when I run the actual script.

> in those cases we renamed the lanes to build_and_upload_{type} to better communicate what they do, and they always do the upload. 

In this case, it doesn't always upload the build, so it still does communicate what it does, it just does something different.

---

What do you think? Should I make the change?

Reply:
I'm not set on it – so feel free to merge as-is!

Reply:
OK, let's merge it for now, but if you feel strongly about it, we can definitely change the naming.

cc @loremattei - would love to hear your take as well.